### PR TITLE
feat(VUEX-section): add vuex docs

### PR DESCRIPTION
**Pull request by deverebor**
Created a section to explain VUEX

### CODE REVIEW

**Comment by tiago-dela-rosa** [comment]
(suggestion) idk if we can call vuex as a middleware, what about "**_Vuex is a state management pattern to Vue applications to centralized state between components_**". Middleware is more like something that works between the process, kind a Auth configuration or a route validation. Nuxt use middlewares btw.

**Comment by deverebor** [comment]
I appreciate your time to help create this section! I believe that this definition that you proposed is even better. My notes are kind of confused hahahaha thankful for the time!

File: README.md [hunk 1]
```diff
@@ -1023,6 +1024,53 @@ export default class MyComponent extends Vue {
...
+`API => Vuex => Components`
+
+![Modelo](./assets/image/modelos.png)
+
+#### VUEX attributes
+
+- state:{} - controls the state;
+- mutations:{} - change the state;
+- actions:{} - actions within a given context;
+- modules:{} - single divisions that contain `states, mutations and actions` watch out for certain modules
```
Comment (tassioFront) [inline README.md:1040]: Suggestion: 

- state:{} - reactive state;
- Getters: {} - responsible for returning specific fields from the state or/and handle them before being readed.
- mutations:{} - responsible for change the state;
- actions:{} - responsible for handle logic and call mutations to change the state;

File: README.md [hunk 2]
```diff
@@ -1023,6 +1024,53 @@ export default class MyComponent extends Vue {
 </script>
 ```
 
+### 8.6 Vuex
+
+Basically, vuex is like a "middleware" it is responsible for being the intermediary between the API and the Components.
```
[the remark this answers is not in the document]
Comment (tassioFront) [reply README.md:1029]: Including, I think is good put doc references here. 😁

File: README.md [hunk 3]
```diff
@@ -1023,6 +1024,53 @@ export default class MyComponent extends Vue {
...
+
+#### Logical order of vuex
+
+![Modulo](./assets/image/logic_order.png)
+
+An **ACTION** must call a **MUTATION** to change a **STATE**;
+
+- **mutations** are synchronous functions, that is, when the system is triggered, it has to wait for the mutation to finish.
+
+- **action** are asynchronous, `async` functions that can be executed and don't need to wait for them to finish.
```
Comment (tassioFront) [inline README.md:1050]: Suggestion: `usually` are asynchronous 

it's not a rule.

File: README.md [hunk 4]
```diff
@@ -1023,6 +1024,53 @@ export default class MyComponent extends Vue {
...
+VUE 3: store -> access to vuex.
+
+anything that changes in vuex will be `'replicated'` in components.
+
+to call an **action** use `store.dispatch`
+to call a **mutation** use `store.commit`
+
+An action receives as a parameter a `context` and a `payload`.
+
+A mutation takes a `state` and a `payload` as a parameter.
```
Comment (tassioFront) [inline README.md:1064]: Question: I do not remember the idea behind this PR, but I'm missing Vuex code snippet examples.  Also, I would not put thoses snippet as we do not use it this way:

```JS
`store.dispatch`
`store.commit`
```

**Comment by deverebor** [comment]
I will close this pr, no more make sense

**Pull request closed without merging**